{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f5447ecc",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "import requests\n",
    "import networkx as nx\n",
    "from networkx.drawing.nx_agraph import graphviz_layout\n",
    "from tqdm.auto import tqdm\n",
    "tqdm.pandas()\n",
    "from IPython.display import display, HTML\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63b69109",
   "metadata": {},
   "source": [
    "# Parse ENA data table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a39bb586",
   "metadata": {},
   "source": [
    "This data was gathered from ENA on 2024-02-13 by using the \"Advanced Search\" with the following options:\n",
    "\n",
    "    Data Type: Raw Reads\n",
    "        Query: tax_tree(2) AND instrument_platform=\"oxford_nanopore\"\n",
    "       Fields: run_accession,experiment_title,tax_id,instrument_platform,base_count,experiment_accession,experiment_alias,instrument_model,isolate,library_construction_protocol,library_gen_protocol,library_layout,library_name,library_selection,library_source,library_strategy,pcr_isolation_protocol,project_name,protocol_label,read_count,run_alias,sample_accession,sample_title,sample_alias,sample_description,sample_material,sequencing_method,sra_bytes,sra_ftp,study_accession,study_title,scientific_name,sub_strain,taxonomic_classification,taxonomic_identity_marker,submitted_format,submitted_bytes,submitted_ftp,submission_accession"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b788770a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('../download/2024-02-13_ENA_results_read_run.tsv', sep=\"\\t\", low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1202987d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "947 ENA runs with .tar.gz ftp data\n"
     ]
    }
   ],
   "source": [
    "# only keep entries with .tar.gz data (likely containing raw fast5 files)\n",
    "df = df.loc[(df.submitted_format == 'OXFORDNANOPORE_NATIVE') & \n",
    "             df.submitted_ftp.str.contains('.tar.gz')]\n",
    "df = df.loc[~df.submitted_ftp.str.lower().str.contains('fastq')]\n",
    "print(len(df), \"ENA runs with .tar.gz ftp data\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4db43dc0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1050bd3d4b124c19be157d1f18fe076f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/947 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get lineage information from ENA taxonomy id\n",
    "#def get_lineage(row):\n",
    "#    try:\n",
    "#        return requests.get(f\"https://www.ebi.ac.uk/ena/taxonomy/rest/tax-id/{row.tax_id}?binomialOnly=true\").json()['lineage']\n",
    "#    except:\n",
    "#        return None\n",
    "def get_lineage(row):\n",
    "    # retrieve current lineage from ncbi \n",
    "    # esearch -db taxonomy -query \"Providencia alcalifaciens\" | efetch -format xml | \n",
    "    # xtract -pattern Taxon -element TaxId -element ScientificName -element Lineage\n",
    "    cmd1 = [\"efetch\", \"-db\", \"taxonomy\", \"-format\", \"xml\", \"-id\", f\"{row.tax_id}\"]\n",
    "    cmd2 = [\"xtract\", \"-pattern\", \"Taxon\", \"-element\", \"Lineage\"]\n",
    "    p1 = subprocess.Popen(cmd1, stdout=subprocess.PIPE, stderr=subprocess.PIPE)\n",
    "    p2 = subprocess.Popen(cmd2, stdin=p1.stdout, stdout=subprocess.PIPE)\n",
    "    out, err = p2.communicate()\n",
    "    out = out.decode().strip()\n",
    "    #print(out)\n",
    "    if not out:\n",
    "        return None\n",
    "    if \";\" not in out:\n",
    "        return None\n",
    "    return out\n",
    "\n",
    "df['lineage'] = df.progress_apply(get_lineage, axis=1)\n",
    "df['lineage'] = df['lineage'].str.rstrip().str.rstrip(';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "db14b301",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run_accession</th>\n",
       "      <th>experiment_title</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>instrument_platform</th>\n",
       "      <th>base_count</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>experiment_alias</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>isolate</th>\n",
       "      <th>library_construction_protocol</th>\n",
       "      <th>...</th>\n",
       "      <th>study_title</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>sub_strain</th>\n",
       "      <th>taxonomic_classification</th>\n",
       "      <th>taxonomic_identity_marker</th>\n",
       "      <th>submitted_format</th>\n",
       "      <th>submitted_bytes</th>\n",
       "      <th>submitted_ftp</th>\n",
       "      <th>submission_accession</th>\n",
       "      <th>lineage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>ERR10447791</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>562</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX9972173</td>\n",
       "      <td>ena-EXPERIMENT-TAB-01-11-2022-07:51:02:527-10527</td>\n",
       "      <td>MinION</td>\n",
       "      <td>SC419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli methylation</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>21338903437</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447791/...</td>\n",
       "      <td>ERA18565060</td>\n",
       "      <td>cellular organisms; Bacteria; Pseudomonadota; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>ERR10447792</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>562</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX9972174</td>\n",
       "      <td>ena-EXPERIMENT-TAB-01-11-2022-07:51:02:528-10529</td>\n",
       "      <td>MinION</td>\n",
       "      <td>SC419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli methylation</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>8465641024</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447792/...</td>\n",
       "      <td>ERA18565060</td>\n",
       "      <td>cellular organisms; Bacteria; Pseudomonadota; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>837</th>\n",
       "      <td>ERR10447796</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>562</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX9972178</td>\n",
       "      <td>ena-EXPERIMENT-TAB-01-11-2022-07:51:02:532-10537</td>\n",
       "      <td>MinION</td>\n",
       "      <td>SC419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli methylation</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>32147641186</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447796/...</td>\n",
       "      <td>ERA18565060</td>\n",
       "      <td>cellular organisms; Bacteria; Pseudomonadota; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>ERR10447797</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>562</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX9972179</td>\n",
       "      <td>ena-EXPERIMENT-TAB-01-11-2022-07:51:02:533-10539</td>\n",
       "      <td>MinION</td>\n",
       "      <td>SC452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli methylation</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>15631095638</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447797/...</td>\n",
       "      <td>ERA18565060</td>\n",
       "      <td>cellular organisms; Bacteria; Pseudomonadota; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>ERR10447803</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>562</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX9972185</td>\n",
       "      <td>ena-EXPERIMENT-TAB-01-11-2022-07:51:02:534-10551</td>\n",
       "      <td>MinION</td>\n",
       "      <td>SC452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli methylation</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>30720455063</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447803/...</td>\n",
       "      <td>ERA18565060</td>\n",
       "      <td>cellular organisms; Bacteria; Pseudomonadota; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34829</th>\n",
       "      <td>ERR10466884</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>309800</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX9988744</td>\n",
       "      <td>ena-EXPERIMENT-TAB-07-11-2022-16:32:01:911-29198</td>\n",
       "      <td>MinION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>rRNA processing in Archaea using Nanopore-base...</td>\n",
       "      <td>Haloferax volcanii DS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>7931920929</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10466884/...</td>\n",
       "      <td>ERA18577947</td>\n",
       "      <td>cellular organisms; Archaea; Euryarchaeota; St...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34830</th>\n",
       "      <td>ERR10466885</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>2285</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX9988745</td>\n",
       "      <td>ena-EXPERIMENT-TAB-07-11-2022-16:32:01:911-29200</td>\n",
       "      <td>MinION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>rRNA processing in Archaea using Nanopore-base...</td>\n",
       "      <td>Sulfolobus acidocaldarius</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>2981578577</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10466885/...</td>\n",
       "      <td>ERA18577947</td>\n",
       "      <td>cellular organisms; Archaea; TACK group; Therm...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34833</th>\n",
       "      <td>ERR10802855</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>309800</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX10251337</td>\n",
       "      <td>ena-EXPERIMENT-TAB-24-01-2023-06:59:00:004-985</td>\n",
       "      <td>MinION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Libraries for PCR-cDNA sequencing were prepare...</td>\n",
       "      <td>...</td>\n",
       "      <td>rRNA processing in Archaea using Nanopore-base...</td>\n",
       "      <td>Haloferax volcanii DS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>10484600688</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR108/ERR10802855/...</td>\n",
       "      <td>ERA20292266</td>\n",
       "      <td>cellular organisms; Archaea; Euryarchaeota; St...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34834</th>\n",
       "      <td>ERR11254698</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>309800</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX10662907</td>\n",
       "      <td>ena-EXPERIMENT-TAB-18-04-2023-18:55:11:693-21786</td>\n",
       "      <td>MinION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>rRNA processing in Archaea using Nanopore-base...</td>\n",
       "      <td>Haloferax volcanii DS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>2898849344</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR112/ERR11254698/...</td>\n",
       "      <td>ERA22056836</td>\n",
       "      <td>cellular organisms; Archaea; Euryarchaeota; St...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34835</th>\n",
       "      <td>ERR11254699</td>\n",
       "      <td>MinION sequencing</td>\n",
       "      <td>309800</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>0</td>\n",
       "      <td>ERX10662908</td>\n",
       "      <td>ena-EXPERIMENT-TAB-18-04-2023-18:55:11:693-21788</td>\n",
       "      <td>MinION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>rRNA processing in Archaea using Nanopore-base...</td>\n",
       "      <td>Haloferax volcanii DS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OXFORDNANOPORE_NATIVE</td>\n",
       "      <td>3066555156</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR112/ERR11254699/...</td>\n",
       "      <td>ERA22056836</td>\n",
       "      <td>cellular organisms; Archaea; Euryarchaeota; St...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>947 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      run_accession   experiment_title  tax_id instrument_platform  \\\n",
       "835     ERR10447791  MinION sequencing     562     OXFORD_NANOPORE   \n",
       "836     ERR10447792  MinION sequencing     562     OXFORD_NANOPORE   \n",
       "837     ERR10447796  MinION sequencing     562     OXFORD_NANOPORE   \n",
       "838     ERR10447797  MinION sequencing     562     OXFORD_NANOPORE   \n",
       "839     ERR10447803  MinION sequencing     562     OXFORD_NANOPORE   \n",
       "...             ...                ...     ...                 ...   \n",
       "34829   ERR10466884  MinION sequencing  309800     OXFORD_NANOPORE   \n",
       "34830   ERR10466885  MinION sequencing    2285     OXFORD_NANOPORE   \n",
       "34833   ERR10802855  MinION sequencing  309800     OXFORD_NANOPORE   \n",
       "34834   ERR11254698  MinION sequencing  309800     OXFORD_NANOPORE   \n",
       "34835   ERR11254699  MinION sequencing  309800     OXFORD_NANOPORE   \n",
       "\n",
       "       base_count experiment_accession  \\\n",
       "835             0           ERX9972173   \n",
       "836             0           ERX9972174   \n",
       "837             0           ERX9972178   \n",
       "838             0           ERX9972179   \n",
       "839             0           ERX9972185   \n",
       "...           ...                  ...   \n",
       "34829           0           ERX9988744   \n",
       "34830           0           ERX9988745   \n",
       "34833           0          ERX10251337   \n",
       "34834           0          ERX10662907   \n",
       "34835           0          ERX10662908   \n",
       "\n",
       "                                       experiment_alias instrument_model  \\\n",
       "835    ena-EXPERIMENT-TAB-01-11-2022-07:51:02:527-10527           MinION   \n",
       "836    ena-EXPERIMENT-TAB-01-11-2022-07:51:02:528-10529           MinION   \n",
       "837    ena-EXPERIMENT-TAB-01-11-2022-07:51:02:532-10537           MinION   \n",
       "838    ena-EXPERIMENT-TAB-01-11-2022-07:51:02:533-10539           MinION   \n",
       "839    ena-EXPERIMENT-TAB-01-11-2022-07:51:02:534-10551           MinION   \n",
       "...                                                 ...              ...   \n",
       "34829  ena-EXPERIMENT-TAB-07-11-2022-16:32:01:911-29198           MinION   \n",
       "34830  ena-EXPERIMENT-TAB-07-11-2022-16:32:01:911-29200           MinION   \n",
       "34833    ena-EXPERIMENT-TAB-24-01-2023-06:59:00:004-985           MinION   \n",
       "34834  ena-EXPERIMENT-TAB-18-04-2023-18:55:11:693-21786           MinION   \n",
       "34835  ena-EXPERIMENT-TAB-18-04-2023-18:55:11:693-21788           MinION   \n",
       "\n",
       "      isolate                      library_construction_protocol  ...  \\\n",
       "835     SC419                                                NaN  ...   \n",
       "836     SC419                                                NaN  ...   \n",
       "837     SC419                                                NaN  ...   \n",
       "838     SC452                                                NaN  ...   \n",
       "839     SC452                                                NaN  ...   \n",
       "...       ...                                                ...  ...   \n",
       "34829     NaN                                                NaN  ...   \n",
       "34830     NaN                                                NaN  ...   \n",
       "34833     NaN  Libraries for PCR-cDNA sequencing were prepare...  ...   \n",
       "34834     NaN                                                NaN  ...   \n",
       "34835     NaN                                                NaN  ...   \n",
       "\n",
       "                                             study_title  \\\n",
       "835                                  E. coli methylation   \n",
       "836                                  E. coli methylation   \n",
       "837                                  E. coli methylation   \n",
       "838                                  E. coli methylation   \n",
       "839                                  E. coli methylation   \n",
       "...                                                  ...   \n",
       "34829  rRNA processing in Archaea using Nanopore-base...   \n",
       "34830  rRNA processing in Archaea using Nanopore-base...   \n",
       "34833  rRNA processing in Archaea using Nanopore-base...   \n",
       "34834  rRNA processing in Archaea using Nanopore-base...   \n",
       "34835  rRNA processing in Archaea using Nanopore-base...   \n",
       "\n",
       "                 scientific_name sub_strain taxonomic_classification  \\\n",
       "835             Escherichia coli        NaN                      NaN   \n",
       "836             Escherichia coli        NaN                      NaN   \n",
       "837             Escherichia coli        NaN                      NaN   \n",
       "838             Escherichia coli        NaN                      NaN   \n",
       "839             Escherichia coli        NaN                      NaN   \n",
       "...                          ...        ...                      ...   \n",
       "34829     Haloferax volcanii DS2        NaN                      NaN   \n",
       "34830  Sulfolobus acidocaldarius        NaN                      NaN   \n",
       "34833     Haloferax volcanii DS2        NaN                      NaN   \n",
       "34834     Haloferax volcanii DS2        NaN                      NaN   \n",
       "34835     Haloferax volcanii DS2        NaN                      NaN   \n",
       "\n",
       "      taxonomic_identity_marker       submitted_format  submitted_bytes  \\\n",
       "835                         NaN  OXFORDNANOPORE_NATIVE      21338903437   \n",
       "836                         NaN  OXFORDNANOPORE_NATIVE       8465641024   \n",
       "837                         NaN  OXFORDNANOPORE_NATIVE      32147641186   \n",
       "838                         NaN  OXFORDNANOPORE_NATIVE      15631095638   \n",
       "839                         NaN  OXFORDNANOPORE_NATIVE      30720455063   \n",
       "...                         ...                    ...              ...   \n",
       "34829                       NaN  OXFORDNANOPORE_NATIVE       7931920929   \n",
       "34830                       NaN  OXFORDNANOPORE_NATIVE       2981578577   \n",
       "34833                       NaN  OXFORDNANOPORE_NATIVE      10484600688   \n",
       "34834                       NaN  OXFORDNANOPORE_NATIVE       2898849344   \n",
       "34835                       NaN  OXFORDNANOPORE_NATIVE       3066555156   \n",
       "\n",
       "                                           submitted_ftp  \\\n",
       "835    ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447791/...   \n",
       "836    ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447792/...   \n",
       "837    ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447796/...   \n",
       "838    ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447797/...   \n",
       "839    ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10447803/...   \n",
       "...                                                  ...   \n",
       "34829  ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10466884/...   \n",
       "34830  ftp.sra.ebi.ac.uk/vol1/run/ERR104/ERR10466885/...   \n",
       "34833  ftp.sra.ebi.ac.uk/vol1/run/ERR108/ERR10802855/...   \n",
       "34834  ftp.sra.ebi.ac.uk/vol1/run/ERR112/ERR11254698/...   \n",
       "34835  ftp.sra.ebi.ac.uk/vol1/run/ERR112/ERR11254699/...   \n",
       "\n",
       "       submission_accession                                            lineage  \n",
       "835             ERA18565060  cellular organisms; Bacteria; Pseudomonadota; ...  \n",
       "836             ERA18565060  cellular organisms; Bacteria; Pseudomonadota; ...  \n",
       "837             ERA18565060  cellular organisms; Bacteria; Pseudomonadota; ...  \n",
       "838             ERA18565060  cellular organisms; Bacteria; Pseudomonadota; ...  \n",
       "839             ERA18565060  cellular organisms; Bacteria; Pseudomonadota; ...  \n",
       "...                     ...                                                ...  \n",
       "34829           ERA18577947  cellular organisms; Archaea; Euryarchaeota; St...  \n",
       "34830           ERA18577947  cellular organisms; Archaea; TACK group; Therm...  \n",
       "34833           ERA20292266  cellular organisms; Archaea; Euryarchaeota; St...  \n",
       "34834           ERA22056836  cellular organisms; Archaea; Euryarchaeota; St...  \n",
       "34835           ERA22056836  cellular organisms; Archaea; Euryarchaeota; St...  \n",
       "\n",
       "[947 rows x 40 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1d4ec1d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['lineage_save'] = df['lineage'].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "58f6d49d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def fix_lineage(lineage):\n",
    "    if lineage is None:\n",
    "        return None\n",
    "    l = []\n",
    "    for lvl in lineage.split(';'):\n",
    "        lvl = lvl.strip()\n",
    "        if lvl == \"cellular organisms\" or 'group' in lvl:\n",
    "            pass\n",
    "        else:\n",
    "            l.append(lvl)\n",
    "    return \"; \".join(l)\n",
    "df['lineage'] = df['lineage_save'].apply(fix_lineage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f9e57f06",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "933 entries with sufficient lineage information\n"
     ]
    }
   ],
   "source": [
    "# only keep entries with sufficient lineage information\n",
    "#\n",
    "df = df.loc[(~df.lineage.isnull()) & (df.lineage.str.split(';').str.len() > 4)]\n",
    "print(len(df), \"entries with sufficient lineage information\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f1afa0d5",
   "metadata": {},
   "outputs": [
    {
     "ename": "ImportError",
     "evalue": "requires pygraphviz http://pygraphviz.github.io/",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/envs/ete3/lib/python3.6/site-packages/networkx/drawing/nx_agraph.py\u001b[0m in \u001b[0;36mpygraphviz_layout\u001b[0;34m(G, prog, root, args)\u001b[0m\n\u001b[1;32m    278\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 279\u001b[0;31m         \u001b[0;32mimport\u001b[0m \u001b[0mpygraphviz\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    280\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mImportError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'pygraphviz'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-11-6d8603fcea53>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     70\u001b[0m     \u001b[0max\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_title\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"ENA Datasets with ONT raw data\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m     \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshow\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 72\u001b[0;31m \u001b[0mplot_lineage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-11-6d8603fcea53>\u001b[0m in \u001b[0;36mplot_lineage\u001b[0;34m(df, lvls)\u001b[0m\n\u001b[1;32m     11\u001b[0m                 \u001b[0mT\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd_edge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mu\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\" group\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mv\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\" group\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 13\u001b[0;31m     \u001b[0mpos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgraphviz_layout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mT\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprog\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"dot\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'-Grankdir=\"LR\"'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     14\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0mnext_nodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/ete3/lib/python3.6/site-packages/networkx/drawing/nx_agraph.py\u001b[0m in \u001b[0;36mgraphviz_layout\u001b[0;34m(G, prog, root, args)\u001b[0m\n\u001b[1;32m    235\u001b[0m     \u001b[0mThis\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0ma\u001b[0m \u001b[0mwrapper\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mpygraphviz_layout\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    236\u001b[0m     \"\"\"\n\u001b[0;32m--> 237\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mpygraphviz_layout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mG\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprog\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mprog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mroot\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mroot\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    238\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    239\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/ete3/lib/python3.6/site-packages/networkx/drawing/nx_agraph.py\u001b[0m in \u001b[0;36mpygraphviz_layout\u001b[0;34m(G, prog, root, args)\u001b[0m\n\u001b[1;32m    279\u001b[0m         \u001b[0;32mimport\u001b[0m \u001b[0mpygraphviz\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    280\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mImportError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 281\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mImportError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"requires pygraphviz \"\u001b[0m \u001b[0;34m\"http://pygraphviz.github.io/\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    282\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mroot\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    283\u001b[0m         \u001b[0margs\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;34mf\"-Groot={root}\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mImportError\u001b[0m: requires pygraphviz http://pygraphviz.github.io/"
     ]
    }
   ],
   "source": [
    "def plot_lineage(df, lvls=6):\n",
    "    T = nx.DiGraph()\n",
    "    for lvl in range(0,lvls):\n",
    "        for val,cnt in df.lineage.str.split(';').str[lvl].value_counts().items():\n",
    "            if val.strip().replace(\" group\", \"\"):\n",
    "                T.add_node(val.strip().replace(\" group\", \"\"), size=cnt)\n",
    "    for lvl in range(0,lvls-1):\n",
    "        for edge,cnt in df.lineage.str.split(';').str[lvl:lvl+2].value_counts().items():\n",
    "            if len(edge) == 2 and None not in edge and \" \" not in edge:\n",
    "                u,v = edge\n",
    "                T.add_edge(u.strip().replace(\" group\", \"\"), v.strip().replace(\" group\", \"\"))\n",
    "\n",
    "    pos = graphviz_layout(T, prog=\"dot\", args='-Grankdir=\"LR\"')\n",
    "\n",
    "    next_nodes = set()\n",
    "    for v in list(T):\n",
    "        if not T.out_edges(v):\n",
    "            for u,v in T.in_edges(v):\n",
    "                next_nodes.add(u)\n",
    "    while len(next_nodes) > 0:\n",
    "        next_nodes_ = set()\n",
    "        for u in next_nodes:\n",
    "            new_y = np.median([pos[v][1] for u,v in T.out_edges(u)])\n",
    "            pos[u] = (pos[u][0], new_y)\n",
    "            for n,_ in T.in_edges(u):\n",
    "                next_nodes_.add(n)\n",
    "        next_nodes = next_nodes_\n",
    "    for v in ['Archaea','Bacteria']:\n",
    "        if v in pos:\n",
    "            pos[v] = (pos[v][0]-90, pos[v][1])\n",
    "    fig,ax = plt.subplots(figsize=(16, 14))\n",
    "    nx.draw_networkx(\n",
    "        T,\n",
    "        pos,\n",
    "        with_labels=False,\n",
    "        node_size=1,\n",
    "        node_color=\"white\",\n",
    "        font_size=11,\n",
    "        font_color=\"Black\",\n",
    "        ax=ax,\n",
    "        connectionstyle='angle',\n",
    "    )\n",
    "    pos_higher = {}\n",
    "    for k, v in pos.items():\n",
    "        pos_higher[k] = (v[0]-140, v[1]+14)\n",
    "    nx.draw_networkx(\n",
    "        T,\n",
    "        pos_higher,\n",
    "        edgelist=[],\n",
    "        with_labels=True,\n",
    "        node_size=0,\n",
    "        node_color=\"white\",\n",
    "        font_size=11,\n",
    "        font_color=\"Black\",\n",
    "        ax=ax,\n",
    "        connectionstyle='angle',\n",
    "    )\n",
    "    nx.draw_networkx(\n",
    "        T,\n",
    "        pos,\n",
    "        with_labels=True,\n",
    "        node_size=np.array([int(i) for i in nx.get_node_attributes(T, \"size\").values()])+200,\n",
    "        labels=nx.get_node_attributes(T, \"size\"),\n",
    "        node_color=\"Grey\",\n",
    "        edgelist=[],\n",
    "        font_size=11,\n",
    "        font_color=\"White\",\n",
    "        ax=ax,\n",
    "    )\n",
    "    ax.set_title(\"ENA Datasets with ONT raw data\")\n",
    "    plt.show()\n",
    "plot_lineage(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2d4819d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# select non-amplified samples\n",
    "sel = (df.library_selection.isin(['RANDOM', 'other', 'unspecified', 'size fractionation'])) & \\\n",
    "      (df.library_source == 'GENOMIC') & \\\n",
    "      (df.library_strategy.isin(['WGS', 'OTHER']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fdd02eeb",
   "metadata": {},
   "outputs": [],
   "source": [
    "GBs = df.submitted_bytes.astype('float') / 10**9\n",
    "fig,ax = plt.subplots(figsize=(4,3))\n",
    "ax.hist(GBs, range=(0,200), bins=100)\n",
    "ax.set(xlabel=\"GB of data\", ylabel=\"# runs per bin\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d2a2e4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "sel &= (GBs > 1) & (GBs < 200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f355f4b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[sel].scientific_name.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1938097d",
   "metadata": {},
   "outputs": [],
   "source": [
    "frequent_sns = df.loc[sel].scientific_name.value_counts().loc[df.loc[sel].scientific_name.value_counts() > 3]\n",
    "non_frequent_sns = df.loc[sel].scientific_name.value_counts().loc[df.loc[sel].scientific_name.value_counts() <= 3]\n",
    "frequent_sns.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10ca7be8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# hand-select\n",
    "pd.set_option('display.max_rows', None)\n",
    "for sn in frequent_sns.index:\n",
    "    display(df.loc[sel & (df.scientific_name == sn)].sort_values(['study_title','submitted_bytes'], ascending=False)[[\n",
    "        'run_accession','study_title', 'project_name', 'sample_title', 'sample_alias', 'sample_description', \n",
    "        'lineage', 'isolate', 'submitted_bytes', 'submitted_ftp']])\n",
    "pd.set_option('display.max_rows', 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6bfa9143",
   "metadata": {},
   "outputs": [],
   "source": [
    "manual_sel = ['ERR7850327','ERR1474981', 'ERR4699901', 'ERR3485550', 'ERR5191772', 'ERR5353275', 'ERR2868914', 'ERR9793808',\n",
    "              'ERR9951991', 'ERR2603589', 'ERR4173917', 'ERR10447806', 'ERR5353220', 'ERR1313294', 'ERR9793809', \n",
    "              'ERR3654117', 'ERR5243236', 'ERR5295516', \n",
    "              'ERR2503988', 'ERR2504003', 'ERR2503999',\n",
    "              'ERR968973', 'ERR1674581', 'ERR2137852',\n",
    "              'ERR3654115', 'ERR5353121', 'ERR5295497',\n",
    "              'ERR5420993', 'ERR5295524', 'ERR5420946',\n",
    "              'ERR5205861', 'ERR5205857',\n",
    "              'ERR1309545', \n",
    "              'ERR5159245', 'ERR5859545',\n",
    "              'ERR4173916', 'ERR4179765',\n",
    "              'ERR5353153', 'ERR5353180',\n",
    "              'ERR11027274', 'ERR11027272',\n",
    "              'ERR7477323', 'ERR7477284',\n",
    "              'ERR3798589', 'ERR3798592', \n",
    "              'ERR4656970', 'ERR4656971',\n",
    "              ]\n",
    "automatic_sel = df.loc[sel & df.scientific_name.isin(non_frequent_sns.index), 'run_accession'].to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "988452d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "sel = df.run_accession.isin(manual_sel) | df.run_accession.isin(automatic_sel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13054fdf",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ds_per_scientific_name = 3\n",
    "#sel_sn = []\n",
    "#for sn in df.loc[sel].scientific_name.unique():\n",
    "#    #sel_sn.extend(df.loc[df.scientific_name == sn].index[:ds_per_scientific_name]) # just take first\n",
    "#    sel_sn.extend(df.loc[df.scientific_name == sn].sort_values('submitted_bytes', ascending=False).index[:ds_per_scientific_name]) # select largest dataset\n",
    "#sel &= df.index.isin(sel_sn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "456e2cff",
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_lineage(df.loc[sel], lvls=6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfe2cf0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "GBs.loc[sel].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e6cf850",
   "metadata": {},
   "outputs": [],
   "source": [
    "sel.sum(), len(df.loc[sel].scientific_name.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "165a1b95",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[:, 'dataset'] = 'ENA'\n",
    "df.loc[:, 'sample'] = df['run_accession']\n",
    "df.loc[:, 'priority'] = 0\n",
    "df.loc[:, 'NAT_source'] = df['submitted_ftp']\n",
    "df.loc[:, 'ref_source'] = ''\n",
    "df.loc[:, 'WGA_source'] = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "70f01fa9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[sel,\n",
    "    ['dataset', 'sample', 'NAT_source', 'ref_source', 'WGA_source', 'priority', \n",
    "     'study_accession', 'experiment_accession', 'sample_accession', 'run_accession', 'submission_accession',\n",
    "     #'experiment_title', 'instrument_platform', 'base_count', 'experiment_alias', 'protocol_label', 'pcr_isolation_protocol', 'sequencing_method', \n",
    "     'study_title', 'project_name', 'sample_title', 'sample_alias', 'sample_description', # 'sample_material', 'run_alias', \n",
    "     'tax_id', 'scientific_name', 'lineage', 'isolate', #'taxonomic_identity_marker',  'sub_strain', 'taxonomic_classification',\n",
    "     'library_construction_protocol', 'library_layout', 'library_name', #'library_selection', 'library_source', 'library_strategy', #'library_gen_protocol',\n",
    "     'instrument_model',  \n",
    "     'submitted_bytes', 'submitted_ftp', #'sra_bytes', 'sra_ftp', 'read_count','submitted_format', \n",
    "    ]].to_csv(\"../datasets/ENA_dataset.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "39ec2f80",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENA\t\t MS061_lactate\tNo684_Glaeser_Mikroplastik\r\n",
      "ENA_dataset.csv  MS061_nitrate\tVeillonella.zip\r\n",
      "MS061_control\t MicrobeMod\tother\r\n"
     ]
    }
   ],
   "source": [
    "! ls \"../datasets/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "02f8dfbb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
